llvm/test/CodeGen/DirectX/flatten-bug-117273.ll

```diff
@@ -0,0 +1,23 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -passes='dxil-flatten-arrays,dxil-op-lower' -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+
+
+@ZerroInitAndUndefArr = internal constant [2 x [3 x float]] [[3 x float] zeroinitializer, [3 x float] undef], align 16
```

Comment:
The test case in #117273 doesn't have anything that should be producing an undef array, and I'm not sure it should be possible to create an undef array in HLSL. Do you know where this is coming from?

Reply:
It wasn't your exact example I just was playing around with how to change the values in  the array after I saw float3(0,0,0) became `zeroinitializer `. I then noticed that if I created an uninitialized array and passed it to the Pallette array I could get an undef array.
 example with fcgl: https://hlsl.godbolt.org/z/oPxxoKf9f
example without: https://hlsl.godbolt.org/z/rr1TocKnj